x-pack/platform/plugins/shared/onechat/CONTRIBUTOR_GUIDE.md

```diff
@@ -0,0 +1,267 @@
+# Agent Builder - contributor guide
+
+This document is intended for platform contributors to the Agent Builder framework,
+explains the base concepts and how to register "platform" tools and agents.
+
+(But please also check the [README.md](./README.md) too for more general information about Agent Builder)
+
+## Difference between built-in tools and agents and their user-created counterparts
+
+Platform and user-created tools and agents share the same concepts and API, but have some notable differences:
+
+### read-only
+
+Platform tools and agents are read-only, and cannot be modified or deleted by the user.
+
+### space awareness
+
+- User-created tools are **space-aware** (accessible exclusively from the space they were created in).
+- Platform tools and agents are **space agnostic**: they are accessible from any space.
+  - _(as long as the user has access to it and the feature is enabled for that space)_
+
+### id namespacing
```

Comment:
suggestion: adding the location of the protected namespace list and how they can add to the namespaces.